net/NetUtil.cpp

```diff
@@ -184,10 +184,41 @@ class DNSResolver
     }
 };
 
-HostEntry resolveDNS(const std::string& addressToCheck)
+HostEntry syncResolveDNS(const std::string& addressToCheck)
 {
+#if !MOBILEAPP
+    // Where we have async DNS then use it for the sync DNS use cases too
+    // so we have a single cache of DNS results.
+    std::mutex mutex;
+    std::condition_variable cv;
+
+    std::shared_ptr<HostEntry> result;
+
+    net::AsyncDNS::DNSThreadDumpStateFn dumpState = [addressToCheck]() -> std::string
+    {
+        std::string state = "syncResolveDNS: [" + addressToCheck + "]";
+        return state;
+    };
+
+    net::AsyncDNS::DNSThreadFn callback = [&mutex, &result, &cv](const HostEntry& hostEntry)
+    {
+        std::unique_lock<std::mutex> lock(mutex);
+        result = std::make_shared<HostEntry>(hostEntry);
+        lock.unlock();
+        cv.notify_one();
+    };
+
+    std::unique_lock<std::mutex> lock(mutex);
+
+    AsyncDNS::lookup(addressToCheck, callback, dumpState);
+
+    cv.wait(lock, [&result]{ return static_cast<bool>(result); });
```

Comment:
No need to cap this with timeout?

Reply:
I don't think so.  Maybe I'll be shown wrong here.